pkg/restapi/issuer/operation/operations.go

```diff
@@ -388,64 +397,55 @@ func didExchangeClient(ariesCtx aries.CtxProvider) (*didexchange.Client, error)
 	return didExClient, nil
 }
 
-func issueCredentialClient(prov issuecredential.Provider) (*issuecredential.Client, error) {
+func issueCredentialClient(prov issuecredential.Provider, actionCh chan service.DIDCommAction) (*issuecredential.Client, error) { // nolint: lll
 	issueCredentialClient, err := issuecredential.New(prov)
 	if err != nil {
 		return nil, err
 	}
 
-	actionCh := make(chan service.DIDCommAction, 1)
-
 	err = issueCredentialClient.RegisterActionEvent(actionCh)
 	if err != nil {
 		return nil, err
 	}
 
-	go issueCredentialActionListener(actionCh)
-
 	return issueCredentialClient, nil
 }
 
-func issueCredentialActionListener(ch <-chan service.DIDCommAction) {
+func (o *Operation) issueCredentialActionListener(ch <-chan service.DIDCommAction) {
 	for msg := range ch {
 		switch msg.Message.Type() {
 		case issuecredsvc.RequestCredentialMsgType:
-			handleRequestCredential(msg)
+			err := o.handleRequestCredential(msg)
+			if err != nil {
+				msg.Stop(fmt.Errorf("handle credential request : %w", err))
+			}
 		default:
 			msg.Stop(fmt.Errorf("unsupported message type : %s", msg.Message.Type()))
 		}
 	}
 }
 
-func handleRequestCredential(msg service.DIDCommAction) {
+func (o *Operation) handleRequestCredential(msg service.DIDCommAction) error {
 	// TODO https://github.com/trustbloc/edge-adapter/issues/124 Validate credential request from wallet
-	issued := time.Now()
+	newDidDoc, err := o.vdriRegistry.Create("peer", vdri.WithServiceEndpoint(o.serviceEndpoint))
```

Comment:
interesting - I thought we weren't going to create new dids for the transaction. Glad to see this included.